File: lulu/flip-bits.py
class Solution:
    """
    @param a, b: Two integer
    return: An integer
    """
    def bitSwapRequired(self, a, b):
        # write your code here
        return self.countOnes(a^b)
        
    def countOnes(self, num):
        # write your code here
        counter = 0
        a = 1
        for i in range(0, 32):
            digit = num & a
            if digit != 0:
                counter += 1
            a *= 2
        return counter

File: joshua/FlipBits.java
package com.lintcode;

/**
 * @author : Joshua
 * Date:     2/20/16
 * @see <a href="http://www.lintcode.com/zh-cn/problem/flip-bits/">将整数A转换为B</a>
 */
public class FlipBits {
    /**
     *@param a, b: Two integer
     *return: An integer
     */
    public static int bitSwapRequired(int a, int b) {
        // write your code here
        return countOnes(a ^ b);

    }

    private static int countOnes(int num) {
        String binaryStr = Integer.toBinaryString(num);
        binaryStr = binaryStr.replaceAll("0", "");
        return binaryStr.length();
    }
}
